TaskManager.java

```diff
@@ -0,0 +1,70 @@
+@Component
+public class TaskManager {
+    private EntityManager entityManager;
+    private static TaskManager instance;
+    private LinkedList<Task> queue = new LinkedList<>();
+    private HashSet<Task> running = new HashSet<>();
+    private HashSet<Task> blocked = new HashSet<>();
+    private Thread checker = new Thread(() -> {
```

Comment:
Executor.newSingleThreadExecutor here? It's convenient if we want to control frequency of checkRun calls